corehq/apps/locations/resources/v0_6.py

```diff
@@ -35,4 +39,68 @@ def dehydrate(self, bundle):
             bundle.data['parent_location_id'] = ''
         bundle.data['location_type_name'] = bundle.obj.location_type.name
         bundle.data['location_type_code'] = bundle.obj.location_type.code
+        bundle.data['location_data'] = bundle.obj.metadata
+        return bundle
+
+    def obj_create(self, bundle, **kwargs):
+        domain = kwargs['domain']
+        if 'name' not in bundle.data:
+            raise BadRequest("'name' is a required field.")
+        if SQLLocation.objects.filter(domain=domain, site_code=bundle.data['site_code']).exists():
+            raise BadRequest("Location on domain with site code already exists.")
+        bundle.obj = SQLLocation(domain=domain)
+        self._update(bundle)
         return bundle
+
+    def obj_update(self, bundle, **kwargs):
+        try:
+            bundle.obj = SQLLocation.objects.get(location_id=kwargs['location_id'])
+        except SQLLocation.DoesNotExist:
+            raise BadRequest("Could not find location with given ID.")
+        if bundle.obj.domain != kwargs.pop('domain'):
+            raise NotFound('Location could not be found.')
+        self._update(bundle)
+        return bundle
+
+    def _update(self, bundle):
+        data = bundle.data
+        # Invalid fields that might be common
+        if data.pop('location_type_name', False):
+            raise BadRequest('Location type name is not editable.')
+        if data.pop('last_modified', False):
+            raise BadRequest('\"last_modified\" field is not editable.')
+
+        easy_field_names = ['name', 'site_code', 'latitude', 'longitude']
+        for field_name in easy_field_names:
+            if field_name in data:
+                setattr(bundle.obj, field_name, data.pop(field_name))
+
+        # Other, less "easy" fields
+        if 'location_data' in data:
+            for key, value in data['location_data'].items():
+                if not isinstance(key, str) or not isinstance(value, str):
```

Comment:
There's further validation on custom fields that can be applied using `LocationFieldsView.get_validator` as here:
https://github.com/dimagi/commcare-hq/blob/bca2d0632f95bbd9bf6601db00dfa5d90d032ec3/corehq/apps/locations/bulk_management.py#L289-L293